newIDE/app/src/guidelines/index.js

```diff
@@ -0,0 +1,214 @@
+import React, { PureComponent } from 'react';
+import { render } from 'react-dom';
+import Button from '@material-ui/core/Button';
+import Popover from '@material-ui/core/Popover';
+import Grid from '@material-ui/core/Grid';
+import Typography from '@material-ui/core/Typography';
+import guidelines from './data.js';
+import Window from '../Utils/Window';
+import { getHelpLink } from '../Utils/HelpLink';
+
+const styles = {
+  container: {
+    maxWidth: 400,
+    padding: 10,
+  },
+  description: {
+    paddingBottom: 10,
+  },
+};
+
+type Props = {|
+  opened: boolean,
+|};
+
+export default class GuidelinePopOver extends PureComponent<Props, State> {
+  _inputRef = React.createRef();
+
+  state = {
+    open: true,
+    index: 0,
+    indexData: 0,
+    indexMax: guidelines.length - 1,
+  };
+
+  togglePopover = () => {
+    this.setState(currentState => ({
+      open: !currentState.open,
+    }));
+  };
+
+  next = () => {
+    this.setState(currentState => {
+      let val = currentState.index;
+      if (val === currentState.indexMax) {
+        val = currentState.indexMax;
+      } else {
+        val = currentState.index + 1;
+      }
+      return {
+        index: val,
+      };
+    });
+  };
+
+  back = () => {
+    this.setState(currentState => {
+      let val = currentState.index;
+      if (val === 0) {
+        val = 0;
+      } else {
+        val = currentState.index - 1;
+      }
+      return {
+        index: val,
+      };
+    });
+  };
+
+  componentDidUpdate(prevProps, prevState) {
+    if (this.state.index !== prevState.index) {
+      this._inputRef.current.updatePosition();
+
+      setTimeout(() => this.setState({ indexData: this.state.index }), 300);
```

Comment:
Any reason behind this timeout?

Reply:
It's for force update the component after debounce, but the position doesn't change.
But it's changing [well here](https://codesandbox.io/s/updateposition-qui-fonctionne-0sqrs) in sandbox.